server/server.js

```diff
@@ -136,30 +144,108 @@ server.get('/api/pregnancy-centers/verify', isLoggedIn, function (req, res) {
 	PregnancyCenterModel.findOne({'verified.address': null}, function (err, pregnancyCenterToVerify) {
 		if (err) {
 			log.error(err)
+			res.boom.badImplementation()
+		} else if (!pregnancyCenterToVerify) {
+			res.boom.notFound()
+		} else {
+			res.status(200).json(pregnancyCenterToVerify)
 		}
-		res.status(200).json(pregnancyCenterToVerify)
+
 	})
 })
 
-server.post('/api/pregnancy-centers', isLoggedIn, function (req, res) {
-	PregnancyCenterModel.create(req.body, (err, result) => {
+server.post('/api/pregnancy-centers', isLoggedInAPI, function (req, res) {
+	Joi.validate(req.body, pregnancyCenterSchemaJoi, {
+		abortEarly: false
+	}, function(err, result) {
 		if (err) {
-			log.error(err)
+			handleJoiValidationError(res, err)
+		} else {
+			PregnancyCenterModel.create(result, function (err, result) {
+				if (err) {
+					log.error(err)
+					res.boom.badImplementation()
+				} else {
+					res.status(201).json(result)
+				}
+
+			})
 		}
-		res.status(201).json(result)
+
 	})
+
+})
+
+server.put('/api/pregnancy-centers/:pregnancyCenterId', isLoggedInAPI, function (req, res) {
```

Comment:
Let's discuss this on Tuesday (I've made a note). It's fine to merge in for now, but I think there are some tweaks that can be made to avoid nested if/else statements and make it slightly more readable 😄